### DISCUSSION TITLE
Symmetry of solutions with multiple levels

### DISCUSSION THREAD

**Discussion by wjwhit** [AMReX-Codes/amrex#3207] [Q&A]
Hello all,

I am working on an implementation of AMReX for compressible flows and noticed that for problems with analytic symmetry, the numerical solution may fail to remain symmetric at late times. When using only a single level, this issue is not present, and when using higher order methods the issue is less noticeable. As such, it seems that the asymmetric placement of boxes within levels causes an asymmetry in the numerical dissipation that eventually results in large scale asymmetries.  Has anyone else noticed this issue, and if so, how did you resolve it? Is there any way to make the placement/sizes of boxes more uniform, for example, around the surface of a sphere? It seems like increasing the number of error buffer cells around a tagged cell helps, although not substantially. We have also experimented with increasing/decreasing the maximum number of cell in a box, although when too small our simulations take excessively long, and when too large the total number of cells is dramatically increased. 

Thank you! 

<img width="937" alt="Screenshot 2023-03-16 at 6 41 52 PM" src="https://user-images.githubusercontent.com/48798126/225768912-f5f7d369-1500-4e66-a955-01f598eebee1.png">

**Comment by zingale** [comment]
I am not aware of a way to enforce that the refinement is perfectly symmetry, but you can increase the grid efficiency to try to help, but that will always defer to blocking_factor, and I don't think it is wise to make blocking_factor smaller than the number of ghost cells you need.

see `amr.grid_eff` here: https://amrex-codes.github.io/amrex/docs_html/GridCreation.html#sec-grid-creation

**Reply by wjwhit** [reply]
Thanks for pointing this out, we'll take a look!

**Comment by drummerdoc** [comment]
For one-off cases, you can always use grid_places to hardcode exactly where you want grids to be, and therefore ensure symmetry yourself, but there is no way that I know of to have the AMR grids created with symmetry.

**Reply by wjwhit** [reply]
We'll give it a try, thank you!
